### ISSUE TITLE
Cannot import data from trakt. 

### ISSUE THREAD

**Issue by rcristi** [IgnisDa/ryot#450]
Hello!
I'm trying to import data from trakt but i'm getting this error:

2023-10-29T19:06:42.790476Z  INFO ryot: Listening on: [::]:8000
thread 'tokio-runtime-worker' panicked at apps/backend/src/importer/trakt.rs:176:68:
called `Result::unwrap()` on an `Err` value: invalid type: null, expected a string at line 1 column 91559
stack backtrace:
   0:     0x7f8a0b942611 - <unknown>
   1:     0x7f8a0a9ef7ec - <unknown>
   2:     0x7f8a0b93f1ae - <unknown>
   3:     0x7f8a0b9423f4 - <unknown>
   4:     0x7f8a0b9492d3 - <unknown>
   5:     0x7f8a0b94901f - <unknown>
   6:     0x7f8a0b9497c6 - <unknown>
   7:     0x7f8a0b9496d4 - <unknown>
   8:     0x7f8a0b948226 - <unknown>
   9:     0x7f8a0b949462 - <unknown>
  10:     0x7f8a0a629213 - <unknown>
  11:     0x7f8a0a6295fa - <unknown>
  12:     0x7f8a0b36fdb4 - <unknown>
  13:     0x7f8a0b393dc0 - <unknown>
  14:     0x7f8a0b362926 - <unknown>
  15:     0x7f8a0b6034c2 - <unknown>
  16:     0x7f8a0b5c0312 - <unknown>
  17:     0x7f8a0b3c2da0 - <unknown>
  18:     0x7f8a0b40344c - <unknown>
  19:     0x7f8a0b3ec4d5 - <unknown>
  20:     0x7f8a0b6bea2b - <unknown>
  21:     0x7f8a0b009431 - <unknown>
  22:     0x7f8a0b3e662e - <unknown>
  23:     0x7f8a0b973760 - <unknown>
  24:     0x7f8a0b973157 - <unknown>
  25:     0x7f8a0b983046 - <unknown>
  26:     0x7f8a0b984f6b - <unknown>
  27:     0x7f8a0b972b43 - <unknown>
  28:     0x7f8a0b9858b2 - <unknown>
  29:     0x7f8a0b97b5b9 - <unknown>
  30:     0x7f8a0b97940c - <unknown>
  31:     0x7f8a0b98a314 - <unknown>
  32:     0x7f8a0b970571 - <unknown>
  33:     0x7f8a0b96fcf3 - <unknown>
  34:     0x7f8a0b96fc0f - <unknown>
  35:     0x7f8a0b974538 - <unknown>
  36:     0x7f8a0b987b71 - <unknown>
  37:     0x7f8a0b984624 - <unknown>
  38:     0x7f8a0b98a2d5 - <unknown>
  39:     0x7f8a0b979af7 - <unknown>
  40:     0x7f8a0b94bf25 - <unknown>

I'm running v3.3.2 (docker:unstable) and my trakt profile is public.

**Comment by IgnisDa** [comment]
Please tell me your trakt username.

**Comment by StefanBerecz** [comment]
Maybe for the same reason, I've got a similar problem, also can't import from trakt, or add any movies manually for that matter
Log of Ryot-Container outputs:
```
2023-10-29T20:14:01.278111Z  INFO ryot: Running version: 3.3.2

2023-10-29T20:14:01.324525Z  INFO ryot: Using database backend: PostgreSQL

2023-10-29T20:14:01.325826Z  INFO sea_orm_migration::migrator: Applying all pending migrations

2023-10-29T20:14:01.334598Z  INFO sea_orm_migration::migrator: No pending migrations

2023-10-29T20:14:01.337984Z  INFO ryot: Using timezone: Redacted

2023-10-29T20:14:01.773213Z  INFO ryot: Listening on: [::]:8000

thread 'tokio-runtime-worker' panicked at apps/backend/src/utils.rs:252:10:

called `Option::unwrap()` on a `None` value

note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
Postgres Output when trying to import or add any movie/show:
```
2023-10-30 01:38:06.182 UTC [743] ERROR:  duplicate key value violates unique constraint "pk-person_to_partial-metadata"

2023-10-30 01:38:06.182 UTC [743] DETAIL:  Key (person_id, partial_metadata_id)=(108, 5480) already exists.

2023-10-30 01:38:06.182 UTC [743] STATEMENT:  INSERT INTO "person_to_partial_metadata" ("person_id", "partial_metadata_id", "relation", "role") VALUES ($1, $2, $3, $4) RETURNING "person_id", "partial_metadata_id", "relation", "role"

2023-10-30 01:38:06.183 UTC [753] ERROR:  duplicate key value violates unique constraint "pk-person_to_partial-metadata"

2023-10-30 01:38:06.183 UTC [753] DETAIL:  Key (person_id, partial_metadata_id)=(108, 5480) already exists.

2023-10-30 01:38:06.183 UTC [753] STATEMENT:  INSERT INTO "person_to_partial_metadata" ("person_id", "partial_metadata_id", "relation", "role") VALUES ($1, $2, $3, $4) RETURNING "person_id", "partial_metadata_id", "relation", "role"

2023-10-30 01:38:06.184 UTC [745] ERROR:  duplicate key value violates unique constraint "pk-person_to_partial-metadata"

2023-10-30 01:38:06.184 UTC [745] DETAIL:  Key (person_id, partial_metadata_id)=(108, 5480) already exists.

2023-10-30 01:38:06.184 UTC [745] STATEMENT:  INSERT INTO "person_to_partial_metadata" ("person_id", "partial_metadata_id", "relation", "role") VALUES ($1, $2, $3, $4) RETURNING "person_id", "partial_metadata_id", "relation", "role"

2023-10-30 01:38:11.486 UTC [743] ERROR:  duplicate key value violates unique constraint "pk-person_to_partial-metadata"

2023-10-30 01:38:11.486 UTC [743] DETAIL:  Key (person_id, partial_metadata_id)=(108, 6454) already exists.

2023-10-30 01:38:11.486 UTC [743] STATEMENT:  INSERT INTO "person_to_partial_metadata" ("person_id", "partial_metadata_id", "relation", "role") VALUES ($1, $2, $3, $4) RETURNING "person_id", "partial_metadata_id", "relation", "role"

2023-10-30 01:38:11.487 UTC [750] ERROR:  duplicate key value violates unique constraint "pk-person_to_partial-metadata"

2023-10-30 01:38:11.487 UTC [750] DETAIL:  Key (person_id, partial_metadata_id)=(108, 6454) already exists.

2023-10-30 01:38:11.487 UTC [750] STATEMENT:  INSERT INTO "person_to_partial_metadata" ("person_id", "partial_metadata_id", "relation", "role") VALUES ($1, $2, $3, $4) RETURNING "person_id", "partial_metadata_id", "relation", "role"
```
Trakt profile is public, database and ryot are a fresh install of the newest stable version too

```
Docker-Compose would be:
version: "3.9"

services:
  postgres:
    image: postgres:15-alpine
    container_name: postgres
    restart: unless-stopped
    volumes:
      - ryot/db:/var/lib/postgresql/data
    environment:
      POSTGRES_PASSWORD: redacted
      POSTGRES_USER: redacted
      POSTGRES_DB: redacted
    network_mode: "ryot_default"

  ryot:
    image: "ghcr.io/ignisda/ryot:latest"
    environment:
      - DATABASE_URL=postgres://redacted:redacted@redacted:5432/postgres
      - USERS_ALLOW_REGISTRATION=false
    ports:
      - "8000:8000"
    pull_policy: always
    container_name: ryot
    network_mode: "ryot_default"
```

**Comment by IgnisDa** [comment]
@StefanBerecz I think yours is a separate issue. Please open a new issue and I'll help you there.

**Comment by rcristi** [comment]
> Please tell me your trakt username.

brainless

**Comment by IgnisDa** [comment]
@rcristi Fixed in `v3.3.3`.

You have a lot of media. I recommend you put `RUST_LOG=ryot=trace` env variable before starting your import so that you can follow the progress.

**Comment by rcristi** [comment]
Now It gives me another error but I see that the data is being imported
`2023-10-30T09:04:26.156096Z TRACE ryot::background: Started job: "AssociatePersonWithMetadata"
2023-10-30T09:04:30.175610Z TRACE ryot::background: Started job: "AssociatePersonWithMetadata"
2023-10-30T09:04:30.262174Z TRACE ryot::background: Started job: "AssociatePersonWithMetadata"
2023-10-30T09:04:30.262563Z TRACE ryot::background: Started job: "AssociatePersonWithMetadata"
2023-10-30T09:04:34.178664Z TRACE ryot::background: Started job: "AssociatePersonWithMetadata"
2023-10-30T09:04:42.209989Z TRACE ryot::background: Started job: "AssociatePersonWithMetadata"
2023-10-30T09:04:42.210806Z TRACE ryot::background: Started job: "AssociatePersonWithMetadata"
2023-10-30T09:05:02.337245Z TRACE ryot::background: Started job: "AssociatePersonWithMetadata"
2023-10-30T09:05:02.355414Z TRACE ryot::background: Started job: "AssociatePersonWithMetadata"
2023-10-30T09:05:02.369261Z TRACE ryot::background: Started job: "AssociatePersonWithMetadata"
2023-10-30T09:05:22.408588Z TRACE ryot::background: Job: "AssociatePersonWithMetadata", Time Taken: 56252ms, Successful = false
thread 'tokio-runtime-worker' panicked at apps/backend/src/miscellaneous/resolver.rs:6527:14:
called `Result::unwrap()` on an `Err` value: ConnectionAcquire(Timeout)
stack backtrace:
   0:     0x7f03841cd681 - <unknown>
   1:     0x7f038327f19c - <unknown>
   2:     0x7f03841ca21e - <unknown>
   3:     0x7f03841cd464 - <unknown>
   4:     0x7f03841d4343 - <unknown>
   5:     0x7f03841d408f - <unknown>
   6:     0x7f03841d4836 - <unknown>
   7:     0x7f03841d4744 - <unknown>
   8:     0x7f03841d3296 - <unknown>
   9:     0x7f03841d44d2 - <unknown>
  10:     0x7f0382eb8cc3 - <unknown>
  11:     0x7f0382eb90aa - <unknown>
  12:     0x7f0383f27df7 - <unknown>
  13:     0x7f0383ee2579 - <unknown>
  14:     0x7f0383dabe81 - <unknown>
  15:     0x7f0383deb69c - <unknown>
  16:     0x7f0383dd46fe - <unknown>
  17:     0x7f0383eae7b4 - <unknown>
  18:     0x7f038390d022 - <unknown>
  19:     0x7f0383dcf1d3 - <unknown>
  20:     0x7f03841fe7d9 - <unknown>
  21:     0x7f03841fe1d0 - <unknown>
  22:     0x7f038420e0bf - <unknown>
  23:     0x7f038420ffe4 - <unknown>
  24:     0x7f03841fdbbc - <unknown>
  25:     0x7f038421092b - <unknown>
  26:     0x7f0384206632 - <unknown>
  27:     0x7f0384204485 - <unknown>
  28:     0x7f038421538d - <unknown>
  29:     0x7f03841fb5ea - <unknown>
  30:     0x7f03841fad6c - <unknown>
  31:     0x7f03841fac88 - <unknown>
  32:     0x7f03841ff5b1 - <unknown>
  33:     0x7f0384212bea - <unknown>
  34:     0x7f038420f69d - <unknown>
  35:     0x7f038421534e - <unknown>
  36:     0x7f0384204b70 - <unknown>
  37:     0x7f03841d6f95 - <unknown>
2023-10-30T09:05:26.440100Z TRACE ryot::background: Job: "AssociatePersonWithMetadata", Time Taken: 96398ms, Successful = false
thread 'tokio-runtime-worker' panicked at apps/backend/src/miscellaneous/resolver.rs:6527:14:
called `Result::unwrap()` on an `Err` value: ConnectionAcquire(Timeout)
stack backtrace:
   0:     0x7f03841cd681 - <unknown>
   1:     0x7f038327f19c - <unknown>
   2:     0x7f03841ca21e - <unknown>
   3:     0x7f03841cd464 - <unknown>
   4:     0x7f03841d4343 - <unknown>
   5:     0x7f03841d408f - <unknown>
   6:     0x7f03841d4836 - <unknown>
   7:     0x7f03841d4744 - <unknown>
   8:     0x7f03841d3296 - <unknown>
   9:     0x7f03841d44d2 - <unknown>
  10:     0x7f0382eb8cc3 - <unknown>
  11:     0x7f0382eb90aa - <unknown>
  12:     0x7f0383f27df7 - <unknown>
  13:     0x7f0383ee2579 - <unknown>
  14:     0x7f0383dabe81 - <unknown>
  15:     0x7f0383deb69c - <unknown>
  16:     0x7f0383dd46fe - <unknown>
  17:     0x7f0383eae7b4 - <unknown>
  18:     0x7f038390d022 - <unknown>
  19:     0x7f0383dcf1d3 - <unknown>
  20:     0x7f03841fe7d9 - <unknown>
  21:     0x7f03841fe1d0 - <unknown>
  22:     0x7f038420e0bf - <unknown>
  23:     0x7f038420ffe4 - <unknown>
  24:     0x7f03841fdbbc - <unknown>
  25:     0x7f038421092b - <unknown>
  26:     0x7f0384206632 - <unknown>
  27:     0x7f0384204485 - <unknown>
  28:     0x7f038421538d - <unknown>
  29:     0x7f03841fb5ea - <unknown>
  30:     0x7f03841fad6c - <unknown>
  31:     0x7f03841fac88 - <unknown>
  32:     0x7f03841ff5b1 - <unknown>
  33:     0x7f0384212bea - <unknown>
  34:     0x7f038420f69d - <unknown>
  35:     0x7f038421534e - <unknown>
  36:     0x7f0384204b70 - <unknown>
  37:     0x7f03841d6f95 - <unknown>
2023-10-30T09:05:26.451230Z TRACE ryot::background: Started job: "AssociatePersonWithMetadata"
2023-10-30T09:05:26.456777Z TRACE ryot::background: Started job: "AssociatePersonWithMetadata"
2023-10-30T09:05:34.399947Z TRACE ryot::background: Job: "AssociatePersonWithMetadata", Time Taken: 64224ms, Successful = false
2023-10-30T09:05:34.415609Z TRACE ryot::background: Started job: "AssociatePersonWithMetadata"
2023-10-30T09:05:34.481496Z TRACE ryot::background: Started job: "AssociatePersonWithMetadata"
2023-10-30T09:05:34.482220Z TRACE ryot::background: Started job: "AssociatePersonWithMetadata"
`

**Comment by IgnisDa** [comment]
I think this is because the database connection is being exhausted. Which database are you using? Please show me your docker compose (or whatever else you're using).

**Comment by rcristi** [comment]
I'm using postgresql. That's my docker-compose.yml:


```
version: "3.9" 
services:
  ryot-postgres:
    container_name: ryot-postgres
    image: postgres:15-alpine
    restart: unless-stopped
    volumes:
      - postgres_storage:/var/lib/postgresql/data
    environment:
      POSTGRES_PASSWORD: ryotpasswd
      POSTGRES_USER: ryotusername
      POSTGRES_DB: ryotdatabase

  ryot:
    image: "ghcr.io/ignisda/ryot:unstable"
    environment:
      - RUST_BACKTRACE=1
      - RUST_BACKTRACE=full
      - RUST_LOG=ryot=trace
      - TZ=Europe/Rome
      - DATABASE_URL=postgres://ryotusername:ryotpasswd@ryot-postgres:5432/ryotdatabase # recommended
      # - DATABASE_URL=sqlite:./ryot-db.sqlite # SQLite database
      # - DATABASE_URL=mysql://mysql:mysql@mysql:6749/mysql # MySQL database
      # - SERVER_INSECURE_COOKIE=true # only needed in localhost or non-https    ports:
      - "8000:8000"
    # volumes:
    # - ./ryot-data:/data # only needed if using sqlite database
    pull_policy: always
    container_name: ryot

volumes:
  postgres_storage:
```

**Comment by IgnisDa** [comment]
Now sure why these errors are coming up. Maybe try restarting both the database and application containers?

**Comment by rcristi** [comment]
> Now sure why these errors are coming up. Maybe try restarting both the database and application containers? Also remove RUST_BACKTRACE env variables. They are not helping since the binary does not have debug symbols.
I tried restarting the containers but nothing changed.  When I get home I'll move everything to another PC and try again.  In the meantime, thank you for your help and sorry to bother you

**Comment by IgnisDa** [comment]
No problem at all. Do LMK if you are able to resolve the problem.
